src/Microsoft.Data.SqlClient/tests/ManualTests/DataCommon/DataTestUtility.cs

```diff
@@ -1068,5 +1071,29 @@ public static string GetMachineFQDN(string hostname)
             }
             return fqdn.ToString();
         }
+
+        public static bool IsRunningAsUWPApp()
```

Comment:
You can cache the resolved value with the first call and use it with consecutive calls.